ingestion/src/metadata/ingestion/ometa/utils.py

```diff
@@ -54,15 +55,19 @@ def get_entity_type(
     return class_name
 
 
-def uuid_to_str(entity_id: Union[str, basic.Uuid]) -> str:
+@singledispatch
+def model_str(arg) -> str:
     """
-    Given an entity_id, that can be a str or our pydantic
-    definition of Uuid, return a proper str to build
-    the endpoint path
-    :param entity_id: Entity ID to onvert to string
-    :return: str for the ID
+    Default model stringifying method
     """
-    if isinstance(entity_id, basic.Uuid):
-        return str(entity_id.__root__)
+    return str(arg)
 
-    return entity_id
+
+@model_str.register(basic.Uuid)
+@model_str.register(basic.FullyQualifiedEntityName)
+@model_str.register(basic.EntityName)
+def _(arg) -> str:
```

Comment:
This lets us have a bit more control on the methods we need to extract __root__ from, as IDK if I'd rather have an if/else checking for an attribute which can hide some issues. Let me know what you think